### PR TITLE
Add PTY module

### PR DESCRIPTION
**Pull request by hargoniX**
I do have a test in my local repository, however I'm not quite sure whether I should add it since it has to spawn a shell as well as (obviously) a PTY in order to run it's methods properly. However that would require us to choose a shell for the tests, for Linux and UNIX based Operating Systems that'd be `sh` I presume? On windows cmd?

If we don't want to mess with such details I can alternatively delete the tests as well, the type signatures aren't all too complex.

### CODE REVIEW

**Comment by soutaro** [comment]
Thank you @hargoniX !

Could you try running `bin/annotate-with-rdoc stdlib/pty/pty.rbs` to import RDoc as markdown. (And delete the `arglists` section.)

File: stdlib/pty/pty.rbs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  # In the non-block form this returns an array of size three,
+  # <tt>[r, w, pid]</tt>.
+  #
+  # In the block form these same values will be yielded to the block:
+  #
+  # +r+:: A readable IO that contains the command's
+  #       standard output and standard error
+  # +w+:: A writable IO that is the command's standard input
+  # +pid+:: The process identifier for the command.
+  def self.getpty: (*String command) -> [IO, IO, Integer]
```
Comment (soutaro) [inline stdlib/pty/pty.rbs:90]: Could you add block accepting patterns?

File: stdlib/pty/pty.rbs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  #		    terminal device is available via +slave_file.path+
+  #
+  # IO#raw! is usable to disable newline conversions:
+  #
+  #   require 'io/console'
+  #   PTY.open {|m, s|
+  #     s.raw!
+  #     ...
+  #   }
+  def self.open: () -> [IO, File]
```
Comment (soutaro) [inline stdlib/pty/pty.rbs:129]: Seems like `open` also accepts block.

File: stdlib/pty/pty.rbs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  # In the non-block form this returns an array of size three,
+  # <tt>[r, w, pid]</tt>.
+  #
+  # In the block form these same values will be yielded to the block:
+  #
+  # +r+:: A readable IO that contains the command's
+  #       standard output and standard error
+  # +w+:: A writable IO that is the command's standard input
+  # +pid+:: The process identifier for the command.
+  def self.spawn: (*String command) -> [IO, IO, Integer]
```
Comment (soutaro) [inline stdlib/pty/pty.rbs:157]: This also accepts block!

**Comment by soutaro** [comment]
For testing, maybe we can use `ruby` itself? (https://github.com/ruby/ruby/blob/master/test/test_pty.rb)

**Comment by hargoniX** [comment]
Regarding the rdoc part, if i run that script it will just clear all docs for me...I suspect that this is due to the fact that pty is a C module? At least that's where i got the docs that are above the functions right now from.

For the rest, I'll get to that!

**Comment by soutaro** [comment]
Okay. I think you installed ruby without docs. (RVM doesn't build RDoc by default. https://rvm.io/rubies/docs)

After you generated RDoc, `$ ri PTY` will show you a doc.

**Comment by hargoniX** [comment]
So I did add my tests, however the block testing confuses me slightly, for example in `test_getpty` I use
```ruby
assert_send_type  "(*::String command) { (::IO r, ::IO w, ::Integer pid) -> untyped } -> untyped",
                   PTY, :getpty, "echo" do |r, w, pid| 1 end
```
Which is the proper type to my understanding of the documentation (https://ruby-doc.org/stdlib-2.7.0/libdoc/pty/rdoc/PTY.html#method-c-getpty) However this assertion tells me that that's apparently not at all what is happening, instead it says I'm getting some sort of array type? So I'm wondering, is the true signature for this method actually
```
(*::String command) { (Array[::IO | ::Integer] args) -> untyped } -> untyped
```
instead of having all three as separate arguments? Or am I missing out on something?

**Comment by soutaro** [comment]
@hargoniX I think the document is incorrect. The implementation passes an array of three objects, not three arguments.

Could you try the type `(*String command) { ([IO, IO, Integer]) -> untyped } -> untyped`? (We may need to fix the doc or the implementation.)

**Comment by hargoniX** [comment]
@soutaro That did indeed do the trick. However looking at the signatures the docs provide I believe this behaviour is kind of predictable since the docs state
```
spawn(command_line) { |r, w, pid| ... }
spawn(command_line) => [r, w, pid]
spawn(command, arguments, ...) { |r, w, pid| ... }
spawn(command, arguments, ...) => [r, w, pid]
```
and as we can see from the 2nd and 4th way to call the method they do return an array so the `|r, w, pid|` syntax they show just unpacks the array already and the docs are fine?

**Comment by soutaro** [comment]
When single array is yielded and the block has multiple parameters, Ruby automatically unpacks the array. So, the examples in the doc works.

We can see the difference when we receive it with single block parameter.

```rb
PTY.spawn("ls") {|x| pp x }
```

**Comment by hargoniX** [comment]
@soutaro Cleaned the history up!

**Pull request merged**